File: app/src/main/java/com/ciastek/todoapp/Task.java
package com.ciastek.todoapp;

public class Task {

    private String summary;
    private String description;
    private TaskPriority priority;
    private TaskState state;
    private int taskId;

    public Task (String summary, String description, TaskPriority priority) {
        this.summary = summary;
        this.description = description;
        this.state = TaskState.NEW;
        this.priority = priority;
    }

    public String getSummary () {
        return summary;
    }

    public void setSummary (String summary) {
        this.summary = summary;
    }

    public String getDescription () {
        return description;
    }

    public void setDescription (String description) {
        this.description = description;
    }

    public TaskPriority getPriority () {
        return priority;
    }

    public void setPriority (TaskPriority priority) {
        this.priority = priority;
    }

    public TaskState getState () {
        return state;
    }

    public void setState (TaskState state) {
        this.state = state;
    }
}

File: app/src/main/java/com/ciastek/todoapp/model/Task.kt
package com.ciastek.todoapp.model

import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey

@Entity(tableName = "tasks")
data class Task constructor(val summary: String,
                            val description: String,
                            @TaskPriority val priority: Int,
                            @PrimaryKey(autoGenerate = true) val id: Long? = null,
                            @TaskState var taskState: Int = STATE.NEW)

object PRIORITY {
    const val MINOR: Int = 0
    const val MAJOR: Int = 1
    const val CRITICAL: Int = 2
}

object STATE {
    const val NEW: Int = 0
    const val IN_PROGRESS: Int = 1
    const val DONE: Int = 2
}
